packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-set-user-vars-accounts-to-reconnect.command.ts

```diff
@@ -0,0 +1,170 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import chalk from 'chalk';
+import { Command, CommandRunner, Option } from 'nest-commander';
+import { Repository } from 'typeorm';
+
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { BillingSubscriptionService } from 'src/engine/core-modules/billing/services/billing-subscription.service';
+import {
+  KeyValuePair,
+  KeyValuePairType,
+} from 'src/engine/core-modules/key-value-pair/key-value-pair.entity';
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+import { WorkspaceCacheVersionService } from 'src/engine/metadata-modules/workspace-cache-version/workspace-cache-version.service';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { CalendarChannelSyncStatus } from 'src/modules/calendar/common/standard-objects/calendar-channel.workspace-entity';
+import { AccountsToReconnectService } from 'src/modules/connected-account/services/accounts-to-reconnect.service';
+import { ConnectedAccountWorkspaceEntity } from 'src/modules/connected-account/standard-objects/connected-account.workspace-entity';
+import { AccountsToReconnectKeys } from 'src/modules/connected-account/types/accounts-to-reconnect-key-value.type';
+import { MessageChannelSyncStatus } from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+
+interface SetUserVarsAccountsToReconnectCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'upgrade-0.23:set-user-vars-accounts-to-reconnect',
+  description: 'Set user vars accounts to reconnect',
+})
+export class SetUserVarsAccountsToReconnectCommand extends CommandRunner {
+  private readonly logger = new Logger(
+    SetUserVarsAccountsToReconnectCommand.name,
+  );
+  constructor(
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+    private readonly billingSubscriptionService: BillingSubscriptionService,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly accountsToReconnectService: AccountsToReconnectService,
+    @InjectRepository(KeyValuePair, 'core')
+    private readonly keyValuePairRepository: Repository<KeyValuePair>,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: SetUserVarsAccountsToReconnectCommandOptions,
+  ): Promise<void> {
+    let activeSubscriptionWorkspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      activeSubscriptionWorkspaceIds = [options.workspaceId];
+    } else {
+      activeSubscriptionWorkspaceIds =
```

Comment:
I think this is deprecated, it's much easier now. Now you should do something like this
```typescript
const workspaces = await this.workspaceRepository.find({
  where: { activationStatus: WorkspaceActivationStatus.ACTIVE, ...(options.workspaceId && { id: options.workspaceId }) },
});
```
It will also allow you to check if the workspace from options exists.